SU2_CFD/include/numerics/CNumerics.hpp

```diff
@@ -459,6 +460,35 @@ class CNumerics {
     TurbPsi_Grad_j = val_turbpsivar_grad_j;
   }
 
+  /*!
+   * \brief Compute the mean rate of strain matrix.
+   * \details The parameter primvargrad can be e.g. PrimVar_Grad_i or Mean_GradPrimVar.
+   * \param[in] nDim - 2 or 3
+   * \param[out] rateofstrain - Rate of strain matrix
+   * \param[in] velgrad - A velocity gradient matrix.
+   */
+  static void ComputeMeanRateOfStrainMatrix(unsigned short nDim, su2double** rateofstrain, const su2double* const* velgrad);
+
```

Comment:
I would generalize this further by defining the functions inline and making them templates, this way you are not constrained by the ** interface and can pass anything for which [][] works. It would save some of the temporary pointers you are creating I think.

Reply:
Thank you for this suggestion. In commit 078427d, I replaced the signature by e.g.

    template<class TWOINDICES_1, class TWOINDICES_2>
    inline static void ComputeStressTensor(unsigned short nDim, TWOINDICES_1& stress, const TWOINDICES_2& velgrad,
          su2double viscosity, su2double density=0.0, su2double turb_ke=0.0, bool reynolds3x3=false)

and moved the body to the h-file. Is this the way you intended?

Yes the temporary arrays of pointers were only meant to convert `su2double[3][3]` into `su2double*[3]`. I removed them now.